### PR TITLE
Test new hot store backward compatibility

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
## Overview
Adds adapted tests from IStoreTest to check whether the new hot store is backward compatible

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3141

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala [hunk 1]
```diff
@@ -791,24 +791,184 @@ trait HotStoreSpec[F[_], M[_]] extends FlatSpec with Matchers with GeneratorDriv
...
+
+        for {
+          _   <- hotStore.putDatum(key, datum)
+          res <- hotStore.getData(key)
+          _   <- S.delay { res should contain theSameElementsAs (Seq(datum)) }
+        } yield ()
+      }
+  }
+
+  it should "append datum if channel already exists" in forAll("channel", "datum") {
```
Comment (odeac) [inline rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala]: this test and the previous one would be better suited to a generative test that puts random datum items in a channel

File: rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala [hunk 2]
```diff
@@ -791,24 +791,184 @@ trait HotStoreSpec[F[_], M[_]] extends FlatSpec with Matchers with GeneratorDriv
...
+
+        for {
+          _   <- store.putContinuation(key, wc)
+          res <- store.getContinuations(key)
+          _   <- S.delay(res shouldBe List(wc))
+        } yield ()
+      }
+    }
+
+  it should "append continuation if channel already exists" in
```
Comment (odeac) [inline rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala]: these two tests can also be written as a generative test

File: rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala [hunk 3]
```diff
@@ -791,24 +791,184 @@ trait HotStoreSpec[F[_], M[_]] extends FlatSpec with Matchers with GeneratorDriv
...
+        val continuation = new StringsCaptor
+        val wc1: WaitingContinuation[Pattern, StringsCaptor] =
+          WaitingContinuation.create(key, patterns, continuation, false)
+        val wc2: WaitingContinuation[Pattern, StringsCaptor] =
+          WaitingContinuation.create(key, List(StringMatch(pattern + 2)), continuation, false)
+
+        for {
+          _   <- store.putContinuation(key, wc1)
+          _   <- store.putContinuation(key, wc2)
+          _   <- store.removeContinuation(key, 0)
```
Comment (odeac) [inline rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala]: why 0 triggers the removal of wc2?

File: rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala (hunk 1 above)
Comment (lukasz-golebiewski) [reply rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala]: Thanks for the review @odeac! There already exist generative tests for `HotStore`. The purpose of this PR is adaptation of the *existing* `IStoreTests` (without changing them too much) and running them against `HotStore` in order to check whether the new impl preserves the behaviour

**Pull request merged**